pkg/i18n/english.go

```diff
@@ -1480,9 +1480,9 @@ func EnglishTranslationSet() TranslationSet {
 		DisabledMenuItemPrefix:              "Disabled: ",
 		NoCommitSelected:                    "No commit selected",
 		NoCopiedCommits:                     "No copied commits",
-		QuickStartInteractiveRebase:         "Quick start interactive rebase",
-		QuickStartInteractiveRebaseTooltip:  "Start an interactive rebase for all commits from HEAD down to the first merge commit or main branch commit.\nIf you would instead like to start an interactive rebase from the selected commit, press {{.editKey}} instead",
-		CouldNotQuickStartInteractiveRebase: "Quick start failed: the HEAD commit is a merge commit or is present on the main branch, so there is no appropriate base commit to start the rebase from.",
+		QuickStartInteractiveRebase:         "Start interactive rebase",
+		QuickStartInteractiveRebaseTooltip:  "Start an interactive rebase for the commits on your branch. This will include all commits from the HEAD commit down to the first merge commit or main branch commit.\nIf you would instead like to start an interactive rebase from the selected commit, press `{{.editKey}}`.",
+		CouldNotQuickStartInteractiveRebase: "Could not start interactive rebase: the HEAD commit is a merge commit or is present on the main branch, so there is no appropriate base commit to start the rebase from. You can start an interactive rebase from a specific commit by selecting the commit and pressing `{{.editKey}}`.",
```

Comment:
It's weird to see a "Disabled:" tooltip in the keybindings menu that starts with "Could not".

Since this is now _only_ used for the disabled reason and not for regular error checking in the handler, it should be rephrased and renamed. I pushed a fixup (6c9796a7384) to do this and to change the logic a bit to make this clearer.

With the show-disabled-reasons-as-toasts branch we will still have the problem that the text is much too long for the bottom line view; also, when used as a disabled tooltip in the menu it doesn't need to be this long, since it basically only repeats what the regular tooltip already says. Not totally sure what to do about this, but my feeling is that the Disabled reason should be shortened a lot to be just "No non-main commits to rebase" or something like that.

Reply:
Thanks for making that commit. There's two separate things I want to respond to here:

## Relying on disabled checks

In the code I see:
> // No need to check for nil here, our DisabledReason callback guarantees it's non-nil

The callback actually does not guarantee this, because a background refresh could change `self.c.Model().Commits` and the commit that was found before may no longer be found (for example it may be that in another process the main branch was checked out). Although this is super unlikely, if it _does_ happen it will cause a panic via nil pointer exception, and I really don't want lazygit to ever panic.

This means that to be safe we need to unfortunately have some convoluted code and double-handling. The proper solution to this problem is to use a mutex to guarantee that the model does not change while a controller action is being processed and if I recall correctly that's something you've looked into recently. Until that work is done, I'm going to play it safe.

## Disabled reasons as toasts

I really don't like the idea of shortening all disabled error messages to fit on one line: some errors are the result of a user accidentally doing something nonsensical, like trying to pull the same branch twice. For that, an error message like 'already pulling this branch' is all you need. But there are some errors that result from a user fundamentally misunderstanding something, and they need help, and that requires a longer error message.

A user might think that 'Start interactive rebase' just picks a convenient base commit somewhat arbitrarily and then get surprised when it fails on the main branch (note that they may have a valid reason to do that e.g. they're just working on a personal project with a single branch). At the very least, they need to be told that if they still want to do an interactive rebase, they can press 'e' on a commit. Expecting the user to go looking for the keybinding tooltip to get that extra info feels wrong. There's also the possibility that the user is on a feature branch but the last commit was a merge, and so our error message would need to capture that too.

In your show-disabled-reasons-as-toasts PR, I mentioned that we could let getDisabledReason function decide whether to show an error as a toast or an error message, and I think this is one such case. Showing 'already rebasing' as a toast is perfect because (almost) nobody believes that you should be able to start a rebase while already rebasing. So in this case we've got two possible error messages, one of which works as a toast and one which doesn't. So we could:
* instead of returning strings from our disabled reason functions, make a DisabledReason struct which contains two fields: a message, and a boolean for whether it's displayed as a toast or a popup, or
* decide whether to render something as a toast based on the length of the message

##

Reply:
I retained your language change but reverted the other parts of the change

Reply:
I see. Too bad that we can't rely on the model not changing concurrently to main thread code (yet).

I made this assumption in other places before, so maybe those need to be audited for the same problem. I can't check myself right now because I'm not at my computer; one case that I remember off the top of my head is `checkSelected` in local_commits_controller.go. In that case it's probably still ok, since for the assumption to not hold someone would have to drop all commits of the repo in another process (I think?).

Anyway, for this PR I have no more objections.